### PR TITLE
Gh569 [ltcmd] Add support for \ShowCommand and \NewCommandCopy

### LINKED ISSUE #569: Improve \ShowCommand
## Brief outline of the enhancement

The recently introduced command `\ShowCommand` should display the payload of commands declared with `\NewDocumentCommand` and friends.

## Minimal example showing the current behaviour

```latex
\RequirePackage{latexbug}       % <--should be always the first line (see CONTRIBUTING)!
\documentclass{article}

\newcommand\foo[1]{payload} \ShowCommand\foo
\renewcommand\foo[1][]{payload} \ShowCommand\foo
\RenewDocumentCommand\foo{o}{payload} \ShowCommand\foo
\RenewDocumentCommand\foo{m}{payload} \ShowCommand\foo

\stop
```

## Minimal example showing the desired new behaviour

Right now the payload is not shown for them but one just sees the argument scanning machinery and the fact that they are protected:
```
? 
> \foo=\protected macro:
->\__cmd_start:nNNnnn {o}\foo  \foo code {\__cmd_grab_D:w []}{}{}.
<argument> \foo 
                
l.6 ...entCommand\foo{o}{payload} \ShowCommand\foo
                                                  
? 
> \foo=\protected macro:
->\__cmd_start_expandable:nNNNNn {m}\foo  \foo  \foo code ?{\__cmd_expandable_g
rab_m:w }.
<argument> \foo 
                
l.7 ...entCommand\foo{m}{payload} \ShowCommand\foo
                                                  
? 
```

### PR DESCRIPTION
**Pull request by PhelypeOleinik**
This PR closes #569.  This adds support for `\ShowCommand` and `\NewCommandCopy` to `ltcmd`.  Commands defined with `\NewDocumentCommand` and `\NewExpandableDocumentCommand` can be fully copied using `\NewDocumentCommand\newcmd\oldcmd`, and they can be shown using, for example, `\ShowCommand\AddToHook`, which prints:
```latex
> \AddToHook=document command:
  #1:m
  #2:o
  #3:+m
->\hook_gput_code:nnn {#1}{#2}{#3}.
```

# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [x] Relevant `\changes` entries in source included
- [x] Relevant `changes.txt` updated
- [x] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: base/doc/ltnews34.tex [hunk 1]
```diff
@@ -286,6 +286,21 @@ \subsection{New \cs{showfloat} command}
...
 
 
+\subsection{Add \pkg{ltcmd} support for \cs{NewCommandCopy} and \cs{ShowCommand}}
+
+Since the 2020-10-01 release (see~\cite{34:ltnews32}), \LaTeX{} provides
+\cs{NewCommandCopy} to copy robust commands, and \cs{ShowCommand} to
+show their definition in the terminal.  In the same release, the
+\pkg{xparse} package was integrated in the kernel (now called
+\pkg{ltcmd}).  But since then, the support for \cs{NewCommandCopy} and
+\cs{ShowCommand} was not implemented in \pkg{ltcmd}.  The present
```
Comment (FrankMittelbach) [inline base/doc/ltnews34.tex:296]: wording: However, the extended support for \cs{...} and \cs{...} were not ...

File: base/ltcmd.dtx [hunk 2]
```diff
@@ -1933,6 +1934,656 @@
...
+%    \end{macrocode}
+%
+% \changes{v1.0h}{2021/08/30}{Added support for \cs{NewCommandCopy}}
+% Since the 2020-10-01 \LaTeXe{} release, support for copying and
+% showing the definition of robust commands is available, but the
+% specifics of each command is implemented separately.  Here we'll add
+% support for copying and showing \pkg{ltcmd} definitions.
+%
+% To fully support copying, we need two commands:  a conditional to test
+% if a command is in fact an \pkg{ltcmd} command, and another to
```
Comment (FrankMittelbach) [inline base/ltcmd.dtx:1951]: a ltcmd command

**Review by FrankMittelbach — APPROVED** [review]
looks fine to me; with some minor niggles (sure Chris will find more ...)

**Comment by car222222** [comment]
But so much has now happened that now Chris has no idea where to find a listing of all the changes.  Is it easy to produce a single diff from (say) 2 months ago? 

He did note that Phelype has fixed some from the list he made some time ago.

**Comment by muzimuzhi** [comment]
A typo in https://github.com/latex3/latex2e/pull/657#issue-722665972: 

> Commands defined with `\NewDocumentCommand` and `\NewExpandableDocumentCommand` can be fully copied using <s>`\NewDocumentCommand`</s> `\NewCommandCopy\newcmd\oldcmd`,

@car222222 Maybe the github doc for [Comparing Commits](https://docs.github.com/en/github/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits) would help.

**Comment by car222222** [comment]
@muzimuzhi Thanks for the pointer.

**Comment by PhelypeOleinik** [comment]
@muzimuzhi  Typo fixed. Thanks!

**Pull request merged**